src/utils/encode.ts

```diff
@@ -81,6 +141,29 @@ export function addHexPrefix(hex: string): string {
  * Prepend or append to string
  *
  * *[internal usage]*
+ *
+ * Pads a string to a certain length with a specific string.
+ * The padding can be applied either to the left or the right of the input string.
+ *
+ * @param {string} str The string to pad.
+ * @param {number} length The target length for the padded string.
+ * @param {boolean} left Set to true to add padding to the left, false to add it to the right.
+ * @param {string} [padding='0'] The string to use for padding. Defaults to '0'.
+ * @returns {string} The padded string.
+ *
+ * @example
+ * ```typescript
+ * const myString = 'hello';
+ * const paddedLeft = padString(myString, 10, true);
+ * console.log(paddedLeft); // Outputs: '00000hello'
+ *
+ * const paddedRight = padString(myString, 10, false);
+ * console.log(paddedRight); // Outputs: 'hello00000'
+ *
+ * // Using a different padding character
+ * const paddedCustom = padString(myString, 10, true, '1');
+ * console.log(paddedCustom); // Outputs: '11111hello'
+ * ```
  */
 function padString(str: string, length: number, left: boolean, padding = STRING_ZERO): string {
```

Comment:
This function is not exported, so JSDoc isn't really useful.
But it's made, so keep it (without `encode.` prefix, as it's not exported).